realm/realm-library/src/main/cpp/io_realm_SyncSession.cpp

```diff
@@ -49,3 +53,47 @@ JNIEXPORT jboolean JNICALL Java_io_realm_SyncSession_nativeRefreshAccessToken(JN
     CATCH_STD()
     return JNI_FALSE;
 }
+
+JNIEXPORT jboolean JNICALL Java_io_realm_SyncSession_nativeWaitForDownloadCompletion(JNIEnv* env,
+                                                                                     jobject session_object,
+                                                                                     jstring localRealmPath)
+{
+    TR_ENTER()
+    try {
+        JStringAccessor local_realm_path(env, localRealmPath);
+        auto session = SyncManager::shared().get_existing_session(local_realm_path);
+
+        if (session) {
+            static jni_util::JavaClass java_sync_session_class(env, "io/realm/SyncSession");
+            static jni_util::JavaMethod java_notify_result_method(
+                env, java_sync_session_class, "notifyAllChangesDownloaded", "(Ljava/lang/Long;Ljava/lang/String;)V");
+            jobject java_session_object = env->NewGlobalRef(session_object);
```

Comment:
Use our wrapper class `JavaGlobalRef` instead, but you need to add copy constructor to the `JavaGlobalRef` class
Then you don't need line 84 and 87-90